testplan/common/utils/process.py

```diff
@@ -37,6 +44,12 @@ def kill_process(proc, timeout=5, signal_=None, output=None):
     :type timeout: ``int``
     :param output: Optional file like object for writing logs.
     :type output: ``file``
+    :param on_failed_termination : ``callable`` or ``None``
+        A callback function that is executed when process fails
+        to terminate after the `timeout`. When supplied, this callback
+        will be executed after SIGTERM fails and before SIGKILL.
+        It receives two arguments: pid as `int` and timeout as `int` and
+        can be leveraged to collect additional diagnostic info about the process.
```

Comment:
Can you please add the rest of the type annotations to the function definition and remove the `type` directives from the docstring? Sort of an ongoing refactoring we do by norm.